src/core.js

```diff
@@ -111,6 +114,53 @@ function github({headers, body}) {
   };
 }
 
+function gitlab({headers, body} = {}) {
+  if (!isGitLabRequestSafe({headers}))
+    throw new UnsafeWebhookPayloadError({
+      provider: {
+        name: 'Gitlab',
+        environmentVariable: 'GITLAB_WEBHOOK_SECRET',
+        webhookLocationInstructions: 'at https://gitlab.com/{OWNERNAME}/{REPONAME}/settings/integrations (swap in the path to your repo)'
+      }
+    });
+
+  if (body.object_kind !== 'merge_request') return {success: false};
+  if (!['opened', 'reopened'].includes(body.object_attributes.state)) return {success: false};
+
+  const {object_attributes: {source, source_branch, last_commit: {id}, target, target_project_id}} = body;
+
+  let {web_url} = source;
+  web_url = url.parse(web_url);
+  const statuses_url = `${web_url.protocol}//${web_url.hostname}/api/v4/projects/${target_project_id}/statuses/${id}`;
+
+  const gitlabApi = axios.create({
+    headers: {
+      'PRIVATE-TOKEN': env.GITLAB_TOKEN
+    }
+  });
+
+  return {
+    ref: source_branch,
+    sha: id,
+    success: true,
+    name: target.path_with_namespace,
+    alias: `https://${source.name.replace(/[^A-Z0-9]/ig, '-')}-${source_branch.replace(INVALID_URI_CHARACTERS, '-')}.now.sh`,
```

Comment:
I think it may be worth it to pull this alias formatting logic out and share it between github/lab.